pkg/client/restconfig/restconfig.go

```diff
@@ -21,108 +21,100 @@ import (
 	"strings"
 	"time"
 
+	"github.com/pkg/errors"
 	"k8s.io/cli-runtime/pkg/genericclioptions"
 	_ "k8s.io/client-go/plugin/pkg/client/auth" // kubectl auth provider plugins
 	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/klog/v2"
 	"sigs.k8s.io/cli-utils/pkg/flowcontrol"
 )
 
 // DefaultTimeout is the default REST config timeout.
 const DefaultTimeout = 5 * time.Second
 
-// A source for creating a rest config
-type configSource struct {
-	name   string                       // The name for the config
-	create func() (*rest.Config, error) // The function for creating the config
-}
-
-// List of config sources that will be tried in order for creating a rest.Config
-var configSources = []configSource{
-	{
-		name:   "podServiceAccount",
-		create: newLocalClusterConfig,
-	},
-	{
-		name:   "kubectl",
-		create: newKubectlConfig,
-	},
-}
-
-// MonoRepoRestClient will attempt to create a new rest config from all
-// configured options and return the first successfully created configuration.
-// It always enables the client-side throttling to fix the commit stuck in the
-// inProgress state issue.
-// It is only used in the legacy mono-repo mode.
-func MonoRepoRestClient(timeout time.Duration) (*rest.Config, error) {
-	return newRestConfig(timeout, true)
-}
-
 // NewRestConfig will attempt to create a new rest config from all configured
 // options and return the first successfully created configuration.
 // The client-side throttling is only determined by if server-side flow control
 // is enabled or not. If server-side flow control is enabled, then client-side
 // throttling is disabled, vice versa.
-// It is used by the multi-repo mode.
 func NewRestConfig(timeout time.Duration) (*rest.Config, error) {
-	return newRestConfig(timeout, false)
-}
-
-func newRestConfig(timeout time.Duration, isMonoRepo bool) (*rest.Config, error) {
 	var errorStrs []string
 
-	for _, source := range configSources {
```

Comment:
I think I may prefer the previous syntax. What was your motivation for this refactor?

Reply:
I needed to expose a constructor that just used the specified path, and I always found this for loop over a package variable of structs to be a convoluted and hard to read way to save a few lines of repeated text. 

The new way does have a slightly deeper cyclomatic complexity but as trade off it shows the fallback options all in one place on the screen at the same time with less indirection and clearer user facing error messages.